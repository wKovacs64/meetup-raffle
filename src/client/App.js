// TODO: remove next line after emotion and theme-ui support automatic runtime
/** @jsxRuntime classic */
/** @jsx jsx */
import '@wkovacs64/normalize.css';
import { jsx, Flex, Container } from 'theme-ui';
import Header from './components/Header';
import Raffle from './components/Raffle';
import AppProviders from './AppProviders';

function App() {
  return (
    <AppProviders>
      <Flex
        sx={{
          flexDirection: 'column',
          height: '100%',
        }}
      >
        <Header />
        <Container as="main">
          <Raffle />
        </Container>
      </Flex>
    </AppProviders>
  );
}

App.displayName = 'App';

export default App;
