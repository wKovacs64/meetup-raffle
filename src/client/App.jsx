import Header from './components/Header';
import Raffle from './components/Raffle';

function App() {
  return (
    <div className="flex h-full flex-col">
      <Header />
      <main className="mx-auto w-full max-w-3xl px-4">
        <Raffle />
      </main>
    </div>
  );
}

App.displayName = 'App';

export default App;
