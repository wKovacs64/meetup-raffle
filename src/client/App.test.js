import React from 'react';
import { render } from 'react-testing-library';
import App from './App';

test('renders without crashing', () => {
  const { container } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
});
