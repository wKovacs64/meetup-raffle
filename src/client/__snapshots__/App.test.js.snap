// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="flex flex-column h-100"
  >
    <header
      class="ph3 pv3 bg-light-red"
    >
      <a
        href="https://github.com/wKovacs64/meetup-raffle"
        class="github-corner"
        aria-label="Open GitHub project"
      >
        <svg
          width="80"
          height="80"
          viewBox="0 0 250 250"
          style="position: absolute; top: 0px; fill: #f4f4f4; right: 0px;"
        >
          <path
            class="octo-banner"
            d="M0 0l115 115h15l12 27 108 108V0z"
            fill="#111"
          />
          <path
            class="octo-arm"
            d="M128 109c-15-9-9-19-9-19 3-7 2-11 2-11-1-7 3-2 3-2 4 5 2 11 2 11-3 10 5 15 9 16"
          />
          <path
            class="octo-body"
            d="M115 115s4 2 5 0l14-14c3-2 6-3 8-3-8-11-15-24 2-41 5-5 10-7 16-7 1-2 3-7 12-11 0 0 5 3 7 16 4 2 8 5 12 9s7 8 9 12c14 3 17 7 17 7-4 8-9 11-11 11 0 6-2 11-7 16-16 16-30 10-41 2 0 3-1 7-5 11l-12 11c-1 1 1 5 1 5z"
          />
        </svg>
      </a>
      <div
        class="dt vh-25-ns center-ns"
      >
        <div
          class="dtc v-mid"
        >
          <h1
            class="f3 f2-m f1-l ma0 lh-title near-white"
          >
            <span
              role="img"
              aria-label="meeting"
            >
              ü§ù
            </span>
            meetup-raffle
            <span
              role="img"
              aria-label="ticket"
            >
              üéüÔ∏è
            </span>
          </h1>
          <h3
            class="f6 f4-m f3-l ma0 lh-title"
          >
            Draw raffle winners at your Meetup event
          </h3>
        </div>
      </div>
    </header>
    <main
      class="flex flex-column flex-grow-1 flex-shrink-0 ph3 w-100 mw6-m mw7-l self-center-ns"
    >
      <form>
        <div
          class="mt3 mb4 mv4-ns"
        >
          <label
            class="f4 f3-ns lh-copy dark-blue"
            for="meetup"
          >
            Meetup name (from your URL):
          </label>
          <input
            name="meetup"
            class="input-reset f4 f3-ns near-black ba bw3 b--moon-gray mt3 pa2 w-100"
            type="text"
            id="meetup"
            placeholder="required"
            required=""
            value=""
          />
        </div>
        <div
          class="mv4"
        >
          <div
            class="mb3"
          >
            <label
              class="f4 f3-ns lh-copy dark-blue"
              for="count"
            >
              Number of winners:
            </label>
          </div>
          <span
            class="flex flex-row"
          >
            <button
              aria-label="decrement"
              type="button"
              disabled=""
              class="bn bg-transparent h3 w3 pointer silver"
              data-testid="decrement-button"
            >
              <svg
                xmlns="http://www.w3.org/2000/svg"
                fill="currentColor"
                viewBox="0 0 40 40"
                preserveAspectRatio="xMidYMid meet"
              >
                <g>
                  <path
                    d="m22.5 17.5v-5h-5v5h-5l7.5 10 7.5-10h-5z"
                  />
                </g>
              </svg>
            </button>
            <input
              id="count"
              class="tc near-black w3 f3 pv1"
              name="count"
              type="text"
              pattern="[0-9]*"
              value="1"
            />
            <button
              aria-label="increment"
              type="button"
              class="bn bg-transparent h3 w3 pointer near-black"
              data-testid="increment-button"
            >
              <svg
                xmlns="http://www.w3.org/2000/svg"
                fill="currentColor"
                viewBox="0 0 40 40"
                preserveAspectRatio="xMidYMid meet"
              >
                <g>
                  <path
                    d="m20 12.5l-7.5 10h5v5h5v-5h5l-7.5-10z"
                  />
                </g>
              </svg>
            </button>
          </span>
        </div>
        <div
          class="mv4"
        >
          <button
            type="button"
            class="link bn bg-transparent pa0 pointer near-black"
            data-testid="advanced-button"
          >
            <span
              class="dib w1"
              role="img"
              aria-label="toggle advanced options"
              data-testid="advanced-button-icon"
            >
              ‚ñ∂
            </span>
            <span
              class="ph2"
            >
              Advanced Options
            </span>
          </button>
          <div
            style="height: 0px; overflow: hidden;"
          >
            <div
              class="mt3"
            >
              <label
                class="f6 f5-ns lh-copy dark-blue"
                for="specificEventId"
              >
                Specific event ID
                <em>
                  (defaults to soonest upcoming/in-progress event)
                </em>
                :
              </label>
              <input
                name="specificEventId"
                class="input-reset f6 f5-ns near-black ba bw1 b--moon-gray mt3 pa2 w-100"
                type="text"
                id="specificEventId"
                placeholder="optional"
                value=""
              />
            </div>
            <div
              class="mt3"
            >
              <label
                class="f6 f5-ns lh-copy dark-blue"
                for="meetupApiKey"
              >
                <a
                  href="https://secure.meetup.com/meetup_api/key/"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  Meetup API key
                </a>
                <em>
                  (enables full names of winners)
                </em>
                :
              </label>
              <input
                name="meetupApiKey"
                class="input-reset f6 f5-ns near-black ba bw1 b--moon-gray mt3 pa2 w-100"
                type="text"
                id="meetupApiKey"
                placeholder="optional"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="mv4"
        >
          <button
            class="db center-ns w-100 w5-ns f5 f4-ns b input-reset ba near-black b--near-black bg-white hover-bg-moon-gray pointer ph5 pv3 shadow-5"
            type="submit"
          >
            Draw
          </button>
        </div>
      </form>
    </main>
  </div>
</div>
`;
